### ISSUE TITLE
MSE calculation between different methods

### ISSUE THREAD

**Issue by leoncesc** [rohanchandra30/Spectral-Trajectory-Prediction#8]
Hi, thanks for sharing your great work.
I have read you code of mse calculation in different methods. It seems that you are normalizing differently in the comparison methods, such as in GRIP([here](https://github.com/rohanchandra30/Spectral-Trajectory-Prediction/blob/8a22e7e2e5d0834925f761ff58642454caa18e6c/comparison_method/GRIP/def_train_eval.py#L172))
`            scaled_train = scale_train ( stream2_out , grip_batch_test)
            mse = MSE(scaled_train/torch.max(scaled_train), grip_batch_test/torch.max(grip_batch_test)) * (torch.max(grip_batch_test)).cpu().detach().numpy()
`
while in the proposed method([here](https://github.com/rohanchandra30/Spectral-Trajectory-Prediction/blob/8a22e7e2e5d0834925f761ff58642454caa18e6c/ours/def_train_eval.py#L471)), mse is calculated in the following way
`    x = np.array(y_gt[:,:, 0])
    x = (x-np.mean(x))/x.max()
    y = np.array(y_gt[:,:, 1])
    y = (y-np.mean(y))/y.max()
    acc = np.power(x-muX, 2) + np.power(y-muY, 2)
    lossVal = np.sum(acc, axis=0)/len(acc)
`
In GRIP, the MSE is rescaled by `torch.max(grip_batch_test)`, while in the proposed method it is not rescaled.

**Comment by rohanchandra30** [comment]
Yes they are normalized differently, but it does not matter. What matters is that the scale of the predictions should match the scale of the ground such that the accuracy can be determined. 

In proposed method, we scaled the ground truth to match the predictions whereas in GRIP, we scale  predictions to match the ground truth. The reason for the approach taken for proposed method was that we were encountering system errors when adopting the same normalization as employed in GRIP.

The grip_batch_test is the ground truth data.

To see why choice of normalization approach doesn't matter, see in [this ](https://github.com/rohanchandra30/Spectral-Trajectory-Prediction/blob/8a22e7e2e5d0834925f761ff58642454caa18e6c/figures/compare.png)table that the results for GRIP and proposed method are on a similar scale.

**Comment by leoncesc** [comment]
@rohanchandra30 thanks for your explanation. But with different scale of mse, are they comparable with each other？
For example, remove the scale factor (torch.max(grip_batch_test) ) below, will the mse of GRIP become as small as proposed method? Why multiply an additional scale factor after mse is calculated in GRIP？
`MSE(scaled_train/torch.max(scaled_train), grip_batch_test/torch.max(grip_batch_test)) * (torch.max(grip_batch_test)).cpu().detach().numpy() `

**Comment by rohanchandra30** [comment]
Ah now I understand the confusion ! :) Yes, now I see that the formulas are written in a confusing way, but it conveys the original goal - to match the scale of the predictions with the ground truth.

If we notice the formulas, we are dividing and then multiplying again by torch.max(...) which sort of cancels each other out. So the net effect is still our original goal. 

So by removing the torch.max multiplier, the scale will be between 0-1 which is incorrect. So you will have to remove both the divider **and** the multiplier to have  similar results.

Note that I have not tested this, this is only what I think would happen. Yes, the division and re-multiplication is, in a way, redundant and we should still expect similar results by removing both division and multiplication. But this would need to be tested.

**Comment by leoncesc** [comment]
thanks again！
In the proposed method, mse is not multiplied with torch.max(label), while it does in GRIP. Is this reasonable?
Further more, do you convert or rescale the predictions to the original coordinate, and calculate ADE and FDE?

**Comment by rohanchandra30** [comment]
> In the proposed method, mse is not multiplied with torch.max(label), while it does in GRIP. Is this reasonable?

Yes, it does not affect results.

> Further more, do you convert or rescale the predictions to the original coordinate, and calculate ADE and FDE?

That's whats happening in GRIP, but reverse is happening in proposed method

**Comment by rohanchandra30** [comment]
Closing this out if no further questions. Feel free to reopen if any :)
